test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+/*!
+ * Copyright 2024 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {beforeEach, afterEach, describe, it} from 'mocha';
+import * as events from 'events';
+import * as sinon from 'sinon';
+import {grpc, Status} from 'google-gax';
+import {Database} from '../src/database';
+import {Session} from '../src/session';
+import {MultiplexedSession} from '../src/multiplexed-session';
+import {Transaction} from '../src/transaction';
+
+class FakeTransaction {
+  options;
+  constructor(options?) {
+    this.options = options;
+  }
+  async begin(): Promise<void> {}
+}
+
+describe('MultiplexedSession', () => {
+  let multiplexedSession;
+
+  function noop() {}
+  const DATABASE = {
+    createSession: noop,
+    databaseRole: 'parent_role',
+  } as unknown as Database;
+
+  let fakeMuxSession;
+  let createSessionStub;
+  const sandbox = sinon.createSandbox();
+
+  const createSession = (name = 'id', props?): Session => {
+    props = props || {};
+
+    return Object.assign(new Session(DATABASE, name), props, {
+      create: sandbox.stub().resolves(),
+      transaction: sandbox.stub().returns(new FakeTransaction()),
+      getMetadata: sandbox.stub().resolves({multiplexed: true}),
+    });
+  };
+
+  const shouldNotBeCalled = sandbox.stub().throws('Should not be called.');
+
+  beforeEach(() => {
+    fakeMuxSession = createSession();
+    createSessionStub = sandbox
+      .stub(DATABASE, 'createSession')
+      .withArgs({multiplexed: true})
+      .callsFake(() => {
+        return Promise.resolve([fakeMuxSession]);
+      });
+    multiplexedSession = new MultiplexedSession(DATABASE);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('instantiation', () => {
+    it('should localize the database instance', () => {
+      assert.strictEqual(multiplexedSession.database, DATABASE);
+    });
+
+    it('should apply defaults', () => {
+      assert.strictEqual(multiplexedSession.refreshRate, 7);
+    });
+
+    it('should create a session object', () => {
+      assert.deepStrictEqual(multiplexedSession._multiplexedSession, null);
+    });
+
+    it('should inherit from EventEmitter', () => {
+      assert(multiplexedSession instanceof events.EventEmitter);
+    });
```

Comment:
nit: Club this instantiation into single test.

Reply:
done